Sming/Components/Network/Arch/Esp32/Platform/StationImpl.cpp

```diff
@@ -420,8 +420,8 @@ void StationImpl::staticScanCompleted(wifi_event_sta_scan_done_t* event, uint8_t
 			wifi_ap_record_t ap_info[number];
 			uint16_t ap_count{0};
 			memset(ap_info, 0, sizeof(ap_info));
-			ESP_ERROR_CHECK(esp_wifi_scan_get_ap_records(&number, ap_info));
 			ESP_ERROR_CHECK(esp_wifi_scan_get_ap_num(&ap_count));
+			ESP_ERROR_CHECK(esp_wifi_scan_get_ap_records(&number, ap_info));
```

Comment:
The call to `esp_wifi_scan_get_ap_num` is actually redundant since it's returned from `esp_wifi_scan_get_ap_records`:

```
memset(ap_info, 0, sizeof(ap_info));
ESP_ERROR_CHECK(esp_wifi_scan_get_ap_records(&number, ap_info));
// TODO: Handle hidden APs
for(unsigned i = 0; i < number; i++) {
list.addElement(new BssInfoImpl(&ap_info[i]));
}
```

Reply:
i was wondering about this and under which conditions the number of detected SSIDs and APs are different, especially, when the number of APs would be less than the number of SSIDs, but since this is the code that the ESP_IDF suggests, I thought it was "the right way" 
I can drop the `esp_wifi_scan_get_ap_num()` call and replace `ap_count` with `number` where necessary (or actually use a variable that is a bit more descriptive)